public/app/features/panel/components/PanelDataErrorView.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { CoreApp, GrafanaTheme2, PanelDataSummary, VisualizationSuggestionsBuilder } from '@grafana/data';
+import { usePanelContext, useStyles2 } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { PanelDataErrorViewProps } from '@grafana/runtime';
+import { CardButton } from 'app/core/components/CardButton';
+import { useDispatch } from 'react-redux';
+import { toggleVizPicker } from 'app/features/dashboard/components/PanelEditor/state/reducers';
+import { changePanelPlugin } from '../state/actions';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+import store from 'app/core/store';
+import { LS_VISUALIZATION_SELECT_TAB_KEY } from 'app/core/constants';
+import { VisualizationSelectPaneTab } from 'app/features/dashboard/components/PanelEditor/types';
+
+export function PanelDataErrorView(props: PanelDataErrorViewProps) {
+  const styles = useStyles2(getStyles);
+  const context = usePanelContext();
+  const builder = new VisualizationSuggestionsBuilder(props.data);
+  const { dataSummary } = builder;
+  const message = getMessageFor(props, dataSummary);
+  const dispatch = useDispatch();
+
+  const openVizPicker = () => {
+    store.setObject(LS_VISUALIZATION_SELECT_TAB_KEY, VisualizationSelectPaneTab.Suggestions);
+    dispatch(toggleVizPicker(true));
+  };
```

Comment:
This is a bit hacky. But tried to make this part of panel edit redux state (like isVizPickerOpen) but required quite a number of changes and in the end it did not feel great either, Ideally, this should be done via URL param. But become too big a change for this PR